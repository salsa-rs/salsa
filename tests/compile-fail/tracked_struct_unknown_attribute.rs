#[salsa::tracked]
struct UnknownAttributeTrackedStruct<'db> {
    #[tracked]
    tracked: bool,
    #[unknown_attr]
    field: bool,
    #[salsa::tracked]
    wrong_tracked: bool,
    /// TestDocComment
    /// TestDocComment
    field_with_doc: bool
}

fn main() {}
