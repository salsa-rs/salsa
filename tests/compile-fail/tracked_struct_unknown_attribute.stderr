error: only a single lifetime parameter is accepted
 --> tests/compile-fail/tracked_struct_unknown_attribute.rs:1:1
  |
1 | #[salsa::tracked]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `salsa::tracked` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot find attribute `unknown_attr` in this scope
 --> tests/compile-fail/tracked_struct_unknown_attribute.rs:5:7
  |
5 |     #[unknown_attr]
  |       ^^^^^^^^^^^^
